### PR TITLE
Add possibility to opt-out from printing step attachments when scenario is not successful

### LINKED ISSUE #1136: Opt out option please: Print attachment step text on error format
So #1041 was merged a while back (@DevSide @charlierudolph), but we have just upgraded to version 4 to see its effect.

Our use case:
- Cucumber used in **e2e tests**.
- After each scenario we attach the **browser logs**.
- In dev mode our app spits a **LOT** of log data (~5 per second); some include JSON dumps.

And so, every time something fails (even `cannot read property of undefined`), we get this **ginormous** print out and have to scroll all the way up to see what the issue was.

----

Would be handy for us to be able to opt-out of this behaviour (or maybe even have it as opt-in?).

It makes certain assumptions with regards to attachments that are not valid with some use-cases.

### PR DESCRIPTION
**Pull request by aurelien-reeves**
# Description

Refs. #1136 

I've added a scenario to show a new `--format-options` that would let the possibility to opt-out from the attachments being printed when a scenario is not successful.

That option is named `printAttachments` and is presumably true per default.

# New contributors, welcome!

This is an easy one, and to help you through it, you'll find all the steps to fix that PR here: https://github.com/cucumber/cucumber-js/blob/17527667f70c64600adf1ac924336c64b65df66e/docs/first-time-contributor-tutorial.md

# Motivation & context

Fixes #1136 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds new behaviour)

# Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](../CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] CHANGELOG has been updated

### CODE REVIEW

**Comment by davidjgoss** [comment]
> That option is named `printStepAttachments` and is presumably true per default.

Given you can attach from hooks and these are also printed:

image

Potentially just `printAttachments` is a better name?

> If the fix is not too complex, I plan to use this PR as a step-by-step tutorial for new contributors.

That sounds really good!

**Comment by aurelien-reeves** [comment]
@mattwynne @16sheep @artismarti any plan to going further with that one?

**Comment by aurelien-reeves** [comment]
@mattwynne @16sheep @artismarti do you plan to continue working on this?
If not, any hard feeling if I take care of it?

**Comment by artismarti** [comment]
> @mattwynne @16sheep @artismarti do you plan to continue working on this? If not, any hard feeling if I take care of it?

None whatsoever, from me! :)

I had kinda forgotten about it, tbh.

**Comment by aurelien-reeves** [comment]
I'll finish implementation for that PR as it would be now difficult for a new comer to continue with this as it has been partially implemented

I hope there won't be hard feelings for this!?

**Comment by artismarti** [comment]
None whatsoever! Thanks for finishing the work :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ Please see [CONTRIBUTING.md](https://github.com/cucumber/cucumber/blob/master/CO
 ### Added
 - Cucumber Expressions now support a wider array of parameter types (see [documentation](https://github.com/cucumber/cucumber-expressions#parameter-types))
 - Improved styling and usability on report from `html` formatter
+- Add a new option to `--format-option`: `printAttachments`.
```
Comment (davidjgoss) [inline CHANGELOG.md:14]: --format-option**s**

File: docs/cli.md [hunk 2]
```diff
@@ -74,6 +74,13 @@ By default, cucumber works in _strict_ mode, meaning it will fail if there are p
 
 See [Parallel](./parallel.md).
 
+## Printing Attachments Details
+
+Printing attachments details can be disabled with
```
Comment (davidjgoss) [inline docs/cli.md:79]: I think this content would sit better in `formatters.md` under the "Options" heading; this doc will shrink over time as we decouple config from the CLI.

File: src/formatter/helpers/issue_helpers.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ export interface IFormatIssueRequest {
   snippetBuilder: StepDefinitionSnippetBuilder
   testCaseAttempt: ITestCaseAttempt
   supportCodeLibrary: ISupportCodeLibrary
+  printAttachments?: Boolean
```
Comment (davidjgoss) [inline src/formatter/helpers/issue_helpers.ts:40]: I don't think it matters at runtime but for consistency could we type as `boolean`?

File: docs/cli.md (hunk 2 above)
Comment (aurelien-reeves) [reply docs/cli.md:79]: I have added some docs in `formatters.md`.
However I think keeping the one in `cli.md` could still be relevant?

Comment (davidjgoss) [reply docs/cli.md:79]: No problem!

**Review by davidjgoss — APPROVED** [review]

**Pull request merged**